src/shed/_codemods.py

```diff
@@ -332,3 +332,49 @@ def _has_none(node):
             return updated_node.with_changes(left=updated_node.right, right=node_left)
         else:
             return updated_node
+
+    @m.call_if_inside(m.Annotation(annotation=m.BinaryOperation()))
+    @m.leave(
+        m.BinaryOperation(
+            left=m.Subscript(value=m.Name(value="Literal")),
+            operator=m.BitOr(),
+            right=m.Name("None"),
+        )
+    )
+    def flatten_literal_op(self, _, updated_node):
+        literal = updated_node.left
+        args = list(literal.slice)
+        for item in args:
+            if m.matches(item, m.SubscriptElement(m.Index(m.Name("None")))):
+                return literal  # Already has "None"
+        args[-1] = args[-1].with_changes(comma=cst.Comma())
+        args.append(
+            cst.SubscriptElement(
+                slice=cst.Index(value=cst.Name(value="None")),
+            )
+        )
+        return literal.with_changes(slice=tuple(args))
+
+    @m.leave(m.Subscript(value=m.Name(value="Union") | m.Name(value="Literal")))
+    def flatten_union_literal_subscript(self, _, updated_node):
+        new_slice = []
+        has_none = False
+        for item in updated_node.slice:
+            if m.matches(item.slice.value, m.Subscript(m.Name("Optional"))):
+                new_slice += item.slice.value.slice  # peel off "Optional"
+                has_none = True
+            elif m.matches(
+                item.slice.value, m.Subscript(m.Name("Union") | m.Name("Literal"))
+            ):
+                new_slice += item.slice.value.slice  # peel off "Union" or "Literal"
```

Comment:
I think this would rewrite `Union[int, Literal["abc"]]` to `Union[int, "abc"]` which is incorrect?  It'd be good to add some test cases for this either way, and handle it by e.g. matching with `item.slice.value, updated_node.value`.